packages/core/src/integration.ts

```diff
@@ -155,3 +164,54 @@ function findIndex<T>(arr: T[], callback: (item: T) => boolean): number {
 
   return -1;
 }
+
+/**
+ * Generate a full integration function from a simple function.
+ * This will ensure to add the given name both to the function definition (as id),
+ * as well as to the integration return value.
+ */
+export function makeIntegrationFn<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  Fn extends (...rest: any[]) => Partial<IntegrationFnResult>,
+>(name: string, fn: Fn): ((...rest: Parameters<Fn>) => ReturnType<Fn> & { name: string }) & { id: string } {
+  const patchedFn = addNameToIntegrationFnResult(name, fn);
+
+  return Object.assign(patchedFn, { id: name });
+}
+
+/**
+ * Convert a new integration function to the legacy class syntax.
+ * In v8, we can remove this and instead export the integration functions directly.
+ *
+ * @deprecated This will be removed in v8!
+ */
+export function convertIntegrationFnToClass<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  Fn extends IntegrationFn<(...rest: any[]) => IntegrationFnResult>,
+>(fn: Fn): IntegrationClass<Integration> {
+  return Object.assign(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    function ConvertedIntegration(...rest: any[]) {
+      const res = {
+        // eslint-disable-next-line @typescript-eslint/no-empty-function
+        setupOnce: () => {},
+        ...fn(...rest),
+        name: fn.id,
+      };
+
+      return res;
+    },
+    { id: fn.id },
+  ) as unknown as IntegrationClass<Integration>;
+}
+
+function addNameToIntegrationFnResult<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  Fn extends (...rest: any[]) => Partial<IntegrationFnResult>,
+>(name: string, fn: Fn): (...rest: Parameters<Fn>) => ReturnType<Fn> & { name: string } {
+  const patchedFn = (...rest: Parameters<Fn>): ReturnType<Fn> & { name: string } => {
+    return { ...fn(...rest), name } as ReturnType<Fn> & { name: string };
```

Comment:
This makes unnecessary copy of fn. You can assign it to a variable and later set the name attribute